tensorboard/errors_test.py

```diff
@@ -36,6 +36,12 @@ def test_with_details(self):
     def test_http_code(self):
         self.assertEqual(errors.InvalidArgumentError().http_code, 400)
 
+    def test_headers(self):
+        e1 = errors.InvalidArgumentError()
+        e2 = errors.InvalidArgumentError()
+        self.assertEmpty(e1.headers)
+        self.assertIsNot(e1.headers, e2.headers)
```

Comment:
Why is it important to check that 2 equivalent errors do not have the exact same headers reference?

Reply:
The way I’ve written it, subclasses add headers by appending to
`self.headers`. (I figured that this was simplest.) If these references
were shared, then in

```python
e1 = errors.UnauthenticatedError(challenge="one")
e2 = errors.UnauthenticatedError(challenge="two")
```

both `e1` and `e2` would have a `headers` list with two entries (one
with challenge `one`, one with challenge `two`) due to the mutation
pollution.

In general, this is just “avoid shared mutable state”. The test fails if
you write the initialization as

```python
class PublicError(RuntimeError):
    http_code = 400
    headers = []
```

rather than

```python
class PublicError(RuntimeError):
    http_code = 400
    def __init__(...):
        self.headers = []
```

even though the former looks reasonable to an untrained eye.